packages/apollo/src/commands/service/__tests__/check.test.ts

```diff
@@ -238,6 +380,130 @@ describe("service:check", () => {
 
   // These are integration tests and not e2e tests because these don't actually hit the remote server.
   describe("integration", () => {
+    describe("federated", () => {
+      describe("should report composition errors correctly", () => {
+        it("vanilla", async () => {
+          captureApplicationOutput();
+          mockCompositionFailure();
+
+          expect.assertions(2);
+
+          await expect(
+            ServiceCheck.run([
+              ...cliKeyParameter,
+              "--serviceName=accounts",
+              `--endpoint=${localURL}/graphql`
+            ])
+          ).rejects.toThrow(
+            federatedServiceCompositionUnsuccessfulErrorMessage
+          );
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+
+        it.skip("--markdown", async () => {
+          captureApplicationOutput();
+          mockCompositionFailure();
+
+          expect.assertions(2);
+
+          await expect(
+            ServiceCheck.run([
+              ...cliKeyParameter,
+              "--serviceName=accounts",
+              `--endpoint=${localURL}/graphql`,
+              "--markdown"
+            ])
+          ).rejects.toThrow(
+            federatedServiceCompositionUnsuccessfulErrorMessage
+          );
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+
+        it.skip("--json", async () => {
+          captureApplicationOutput();
+          mockCompositionFailure();
+
+          expect.assertions(2);
+
+          await expect(
+            ServiceCheck.run([
+              ...cliKeyParameter,
+              "--serviceName=accounts",
+              `--endpoint=${localURL}/graphql`,
+              "--json"
+            ])
+          ).rejects.toThrow(
+            federatedServiceCompositionUnsuccessfulErrorMessage
+          );
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+      });
+
+      describe("should report composition success correctly", () => {
+        it("vanilla", async () => {
+          captureApplicationOutput();
+          mockCompositionSuccess();
+
+          expect.assertions(2);
+
+          await expect(
+            ServiceCheck.run([
+              ...cliKeyParameter,
+              "--serviceName=accounts",
+              `--endpoint=${localURL}/graphql`
+            ])
+          ).resolves.not.toThrow();
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+
+        it.skip("--markdown", async () => {
```

Comment:
Should this be skipped?

Reply:
Yes, same as https://github.com/apollographql/apollo-tooling/pull/1308#discussion_r291298332:

>Yes it should because `--markdown` isn't implemented yet. I'll leave a comment here and at the other skip sites with the ticket that'll implementing this.